api/src/repositories/draft-repository.ts

```diff
@@ -0,0 +1,121 @@
+import { QueryResult } from 'pg';
+import SQL, { SQLStatement } from 'sql-template-strings';
+import { z } from 'zod';
+import { ApiExecuteSQLError } from '../errors/api-error';
+import { BaseRepository } from './base-repository';
+
+export const WebformDraftList = z.object({
+  webform_draft_id: z.number(),
+  name: z.string(),
+  data: z.object({}),
+  create_date: z.date(),
+  update_date: z.date().nullable()
+});
+
+export type WebformDraftList = z.infer<typeof WebformDraftList>;
+
+/**
+ * A repository class for accessing draft data.
+ *
+ * @export
+ * @class DraftRepository
+ * @extends {BaseRepository}
+ */
+export class DraftRepository extends BaseRepository {
+  async deleteDraft(draftId: number): Promise<QueryResult> {
```

Comment:
I think `QueryResults` can take a generic value representing the actual return value?

Reply:
not sure what the basic delete returns so I modified it to return the id of the deleted item